server/src/main/java/org/elasticsearch/transport/ConnectTransportException.java

```diff
@@ -41,6 +42,16 @@ public ConnectTransportException(StreamInput in) throws IOException {
         }
     }
 
+    /**
+     * The ES REST API is a gateway to a single or multiple clusters. If there is an error connecting to other servers, then we should
+     * return a retryable 502 BAD_GATEWAY error, instead of the parent class' non-retryable 500 INTERNAL_SERVER_ERROR response code.
```

Comment:
Wording nit: there is no absolute concept of "retryable" for HTTP status codes. Clients generally have some heuristics about whether to retry on a failure, and 502s tend to be retried, but this is ultimately up to the client to decide.

```suggestion
      * The ES REST API is a gateway to a single or multiple clusters. If there is an error connecting to other servers, then we should
      * return a 502 BAD_GATEWAY status code instead of the parent class' 500 INTERNAL_SERVER_ERROR. Clients tend to retry on a 502 but not
      * on a 500, and retrying may help on a connection error.
```